cmd/firectl/main.go

```diff
@@ -218,16 +212,38 @@ func main() {
 	vmmCtx, vmmCancel := context.WithCancel(ctx)
 	defer vmmCancel()
 
-	cmd, cleanup, err := buildCommand(vmmCtx, fcCfg.SocketPath, opts)
-	if err != nil {
-		log.Fatalf("Failed to build command: %v", err)
+	machineOpts := []firecracker.Opt{
+		firecracker.WithLogger(log.NewEntry(logger)),
 	}
-	defer cleanup()
 
-	m, err := firecracker.NewMachine(fcCfg,
-		firecracker.WithLogger(log.NewEntry(logger)),
-		firecracker.WithProcessRunner(cmd),
-	)
+	if len(opts.FcBinary) != 0 {
+		finfo, err := os.Stat(opts.FcBinary)
+		if os.IsNotExist(err) {
+			log.Fatalf("Binary, %q, does not exist: %v", opts.FcBinary, err)
+		}
+
+		if err != nil {
+			log.Fatalf("Failed to stat binary, %q: %v", opts.FcBinary, err)
+		}
+
+		if finfo.IsDir() {
+			log.Fatalf("Binary, %q, is a directory", opts.FcBinary)
+		} else if finfo.Mode()&0111 == 0 {
```

Comment:
nit: I'd prefer to have the mask as a constant.

Reply:
I'm not sure we need this checks at all. Why not just to rely on `exec.Command` ?

Reply:
The errors returned from `exec.Command` can be a little ambiguous. This is to be more specific. For instance if the binary is a directory, `exec.Command` will return "permission denied."